mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -104,10 +104,14 @@ static FailureOr<Operation *> getCompressedMaskOp(OpBuilder &rewriter,
   if (createMaskOp) {
     OperandRange maskOperands = createMaskOp.getOperands();
     size_t numMaskOperands = maskOperands.size();
+    // The `vector.create_mask` op creates a mask arrangement without any zeros
+    // at the front. Also, because `numFrontPadElems` is strictly smaller than
+    // `numSrcElemsPerDest`, the compressed mask generated by shifting the
+    // original mask by `numFrontPadElems` will not have any zeros at the front
+    // as well.
```

Comment:
I don't quite follow this comment. Is it meant to explain where the definition of `s0` is coming from? Or, more specifically:

> the compressed mask generated by shifting the original mask

where does shifting happen?

Reply:
The shifting happens when you pad `numFrontPadElems` to the mask so each individual mask value points to its correct element.

The shifting (to the right) is done by adding `numFrontPadElems` to the `ceildiv` in the affine expression. namely:
`s0 = (s0 + numFrontPadElems).ceilDiv(numSrcElemsPerDest);`